### ISSUE TITLE
AES encrypted files sometimes produce wrong checksums

### ISSUE THREAD

**Issue by philipparndt** [srikanth-lingala/zip4j#22]
Hi,

when I create a password protected zip file I get the following error message during uncompress:
```
net.lingala.zip4j.exception.ZipException: java.io.IOException: Reached end of data for this entry, but aes verification failed
	at net.lingala.zip4j.tasks.AsyncZipTask.performTaskWithErrorHandling(AsyncZipTask.java:48)
	at net.lingala.zip4j.tasks.AsyncZipTask.execute(AsyncZipTask.java:35)
	at net.lingala.zip4j.ZipFile.extractAll(ZipFile.java:431)
```

when compressing files with a specific compressed size. See 
[zip4j-failing-compress.zip](https://github.com/srikanth-lingala/zip4j/files/3386075/zip4j-failing-compress.zip) for an example (Maven project with failing unit test).

I guess the issue occurs when `net.lingala.zip4j.io.outputstream.CompressedOutputStream.decrementBytesWrittenForThisEntry(int)` is called. Maybe the length is calculated correct but the checksum is using a wrong range.

The issue is reproducible when the stacktrace runs through `net.lingala.zip4j.io.outputstream.DeflaterOutputStream.deflate()` and the `len` is in between 1..3

The resulting zip file can be decompressed, but throws an error when using zip4j, 7z and winrar.

**Comment by srikanth-lingala** [comment]
I appreciate you providing me with a reproducible sample. It greatly helps to investigate the issue. I will look into it and update this thread.

**Comment by srikanth-lingala** [comment]
Fixed in v2.1.1 which is released today

**Comment by philipparndt** [comment]
thank you for fixing this issue. I've tested it in my environment and it looks great!
